File: java/src/main/java/com/azure/data/cosmos/serialization/hybridrow/recordio/Segment.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.data.cosmos.serialization.hybridrow.recordio;

public final class Segment {

    private final String comment;
    private final int length;
    private final String sdl;

    public Segment(String comment, String sdl) {
        this.comment = comment;
        this.sdl = sdl;
        this.length = 0;
    }

    public String comment() {
        return this.comment;
    }

    public int length() {
        return this.length;
    }

    public String sdl() {
        return this.sdl;
    }
}

File: src/Serialization/HybridRow/RecordIO/Segment.cs
// ------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All rights reserved.
// ------------------------------------------------------------

#pragma warning disable CA1051 // Do not declare visible instance fields

namespace Microsoft.Azure.Cosmos.Serialization.HybridRow.RecordIO
{
    using System;
    using Microsoft.Azure.Cosmos.Serialization.HybridRow.Schemas;

    public struct Segment
    {
        public int Length;
        public string Comment;
        public string SDL;
        public Namespace Schema;

        [Obsolete("Use object-model constructor instead.")]
        public Segment(string comment, string sdl)
        {
            this.Length = 0;
            this.Comment = comment;
            this.SDL = sdl;
            this.Schema = null;
        }

        public Segment(string comment, Namespace ns)
        {
            this.Length = 0;
            this.Comment = comment;
            this.SDL = null;
            this.Schema = ns;
        }
    }
}
